File: backend/src/main/scala/microservices/operations/hotel/objects/CreateManagerRoomTypePlannerRequest.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateManagerRoomTypePlannerRequest(
    managerId: String,
    roomTypeName: String,
    capacity: Int,
    bedType: String,
    nightlyPrice: String,
    currency: String,
    availableRooms: Int,
    inventoryStartDate: String,
    inventoryEndDate: String,
    roomImageUrl: Option[String]
)
object CreateManagerRoomTypePlannerRequest:
  given sourceEncoder: Encoder[CreateManagerRoomTypePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateManagerRoomTypePlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/hotel/objects/CreateManagerRoomTypePlannerRequest.ts
export type CreateManagerRoomTypePlannerRequest = {
  managerId: string
  roomTypeName: string
  capacity: number
  bedType: string
  nightlyPrice: string
  currency: string
  availableRooms: number
  inventoryStartDate: string
  inventoryEndDate: string
  roomImageUrl?: string | null
}

export const createManagerRoomTypePlannerRequestFromJson = (json: string): CreateManagerRoomTypePlannerRequest =>
  JSON.parse(json) as CreateManagerRoomTypePlannerRequest

export const createManagerRoomTypePlannerRequestToJson = (value: CreateManagerRoomTypePlannerRequest): string =>
  JSON.stringify(value)
